crates/subspace-networking/src/node_runner.rs

```diff
@@ -690,22 +687,31 @@ where
         let local_peer_id = *self.swarm.local_peer_id();
 
         if let IdentifyEvent::Received { peer_id, mut info } = event {
-            debug!(?peer_id, protocols=?info.protocols, "IdentifyEvent::Received");
+            debug!(?peer_id, protocols = ?info.protocols, "IdentifyEvent::Received");
 
             // Check for network partition
             if info.protocol_version != self.protocol_version {
                 debug!(
                     %local_peer_id,
                     %peer_id,
-                    local_protocol_version=%self.protocol_version,
-                    peer_protocol_version=%info.protocol_version,
-                    "Peer has different protocol version. Peer was banned.",
+                    local_protocol_version = %self.protocol_version,
+                    peer_protocol_version = %info.protocol_version,
+                    "Peer has different protocol version, banning temporarily",
                 );
 
-                self.ban_peer(peer_id);
+                self.temporary_bans.lock().create_or_extend(&peer_id);
+
                 return;
             }
 
+            // Remove temporary ban if there was any
+            self.temporary_bans.lock().remove(&peer_id);
+            // Forget about this peer until they upgrade
+            let _ = self.swarm.disconnect_peer_id(peer_id);
```

Comment:
Incorrect snippet placing.

Reply:
I guess you wanted to put this snippet near the actual bans and before the `return`

Reply:
Hm... indeed :man_facepalming: